apps/pie-docs/src/_11ty/shortcodes/indexPageDisplay.js

```
@@ -0,0 +1,73 @@
+ * @param {string} itemKey - Is the key of the element in the main menu, and the page we will be rendering e.g. components
+ * @param {string} excludedElements - A list of keys from pages that have to be excluded e.g. ['Component status']
+ * @param {string} imageSrcDirectory - Optional when need to replace the default directory
+ * @returns {string}  a <div> element containing a list of cards representing all elements under the itemKey category
+ */
+const defaultImageDirectory = '../../assets/img/index';
+const toSlug = (string) => string.toLowerCase().replace(/\s+/g, '-');
+
+const getDraftPagesList = (collection) => {
+    const draftPages = [];
+    collection.forEach((item) => {
+        if (item.data && item.data.draft) draftPages.push(item.data.title);
+    });
+    return draftPages;
+};
+
+module.exports = function ({
+    collection,
+    itemKey,
+    excludedElements,
+    imageSrcDirectory,
+}) {
+    const menuItems = find(collection, itemKey);
+    const draftPages = getDraftPagesList(collection);
+
+    const indexElements = menuItems.map((element) => {
+        if (!excludedElements.includes(element.title) && !draftPages.includes(element.title)) {
+            const menuItemSlug = toSlug(element.title);
+            const itemKeySlug = toSlug(itemKey);
+            const imageDirectory = imageSrcDirectory || defaultImageDirectory;
+            const imgSrc = `${imageDirectory}/${itemKeySlug}/${menuItemSlug}.svg`;
+            const imgMobileSrc = `${imageDirectory}/${itemKeySlug}/${menuItemSlug}-mobile.svg`;
+            const fallbackImage = `${imageDirectory}/${itemKeySlug}/default.svg`;
+
+            const hasSource = fs.existsSync(path.join(__dirname, imgSrc));
+            const hasMobileSource = fs.existsSync(path.join(__dirname, imgMobileSrc));
+
+            const renderFallbackImage = () => {
+                console.warn(`Fallback image used for ${menuItemSlug}. Please ensure image is provided.`);
```

I would prefer we fail the build here. Unless we are happy with a fallback image in prod?

---

I added because it improved the experience while building but I think it should not be too difficult to build new pages and I rather throw an error indeed. I'll change this to throw.